### PR TITLE
Removed resize function and added support for animated GIF

### PR DESCRIPTION
**Pull request by beautylix**
I changed the code, removed the resize function, and added support for animated GIF image, please review and merge it.

### CODE REVIEW

File: start.php [hunk 1]
```diff
@@ -159,6 +159,9 @@ function ckeditor_addons_page_handler($segments) {
 			}
 
 			header("Content-type: image/jpeg");
+			header("Content-type: image/jpg");
```
Comment (hypeJunction) [inline start.php:157]: You can't send all 3 headers. Need to detect correct mkme type.

File: .gitattributes [hunk 2]
```diff
@@ -1,21 +0,0 @@
-# Auto detect text files and perform LF normalization
-*        text=auto
```
Comment (hypeJunction) [inline .gitattributes:2]: Why is this file deleted?

File: .gitignore [hunk 3]
```diff
@@ -1,11 +0,0 @@
-#Ignore files that will need to be changed locally
-.DS_Store
```
Comment (hypeJunction) [inline .gitignore:2]: Why is this file deleted?

File: actions/ckeditor/upload.php [hunk 4]
```diff
@@ -28,7 +28,7 @@
 
 $file = new ElggFile();
 $file->owner_guid = $user->guid;
-$file->setFilename("ckeditor/{$hash}.jpg");
+$file->setFilename("ckeditor/$hash");
```
Comment (hypeJunction) [inline actions/ckeditor/upload.php:33]: Files need to have a proper extension.

File: actions/ckeditor/upload.php [hunk 5]
```diff
@@ -16,8 +16,8 @@
 
 $user = elgg_get_logged_in_user_entity();
 
-$max_width = (int) elgg_get_plugin_setting('upload_max_width', 'ckeditor_addons', 600);
-$resized = get_resized_image_from_existing_file($_FILES['upload']['tmp_name'], $max_width, $max_width);
+$resized = get_uploaded_file('upload');
```
Comment (hypeJunction) [inline actions/ckeditor/upload.php:19]: You need to sniff mime here to keep gifs as is and resize the rest.

File: views/default/plugins/ckeditor_addons/settings.php [hunk 6]
```diff
@@ -77,12 +77,3 @@
 	));
 	?>
 </div>
-<div>
```
Comment (hypeJunction) [inline views/default/plugins/ckeditor_addons/settings.php:80]: Removing a feature is not a fix to your problem

File: views/default/resources/ckeditor/browse.php [hunk 7]
```diff
@@ -14,31 +14,6 @@
 
 $dir = new \DirectoryIterator($ck_dir);
 
-foreach ($dir as $file) {
```
Comment (hypeJunction) [inline views/default/resources/ckeditor/browse.php:17]: This prevents users from selecting any files they have uploaded before.

File: start.php (hunk 1 above)
Comment (beautylix) [inline start.php:157]: I thought so, but when I added this, and tested it, that worked well

File: .gitattributes (hunk 2 above)
Comment (beautylix) [inline .gitattributes:2]: I thought it was redundant , maybe I am wrong

File: .gitignore (hunk 3 above)
Comment (beautylix) [inline .gitignore:2]: I thought it was redundant , maybe I am wrong

File: actions/ckeditor/upload.php (hunk 4 above)
Comment (beautylix) [inline actions/ckeditor/upload.php:33]: uploaded file extension is reserved, just not showing, when you click the image to save to your local device, you will see the extension, so no needs to add the jpg

File: actions/ckeditor/upload.php (hunk 5 above)
Comment (beautylix) [inline actions/ckeditor/upload.php:19]: I do not want resize function, because somebody will highlight the whole image to view the detail,  especially if somebody uploaded a image including many small letters, if you resized it ,it would be blur

File: views/default/plugins/ckeditor_addons/settings.php (hunk 6 above)
Comment (beautylix) [inline views/default/plugins/ckeditor_addons/settings.php:80]: Because the resize function has been removed, the setting is not needed.

File: views/default/resources/ckeditor/browse.php (hunk 7 above)
Comment (beautylix) [inline views/default/resources/ckeditor/browse.php:17]: It just processes the gif and png image here, now all image types have the same process, and so I deleted the codes.

**Comment by beautylix** [comment]
A key point, elgg stores data outside rootwww, even a JavaScript file got uploaded, it would not hack the website, right?

**Comment by hypeJunction** [comment]
Thanks for the effort, but I wasn't able to pull this in, as it was not complete. See https://github.com/hypeJunction/Elgg-ckeditor_addons/commit/0243794178dc9bb1e7c588110ad63f1e5f1e11ae
